protocols/kad/src/protocol_priv.rs

```diff
@@ -33,16 +33,16 @@ use bytes::BytesMut;
 use codec::UviBytes;
 use futures::prelude::*;
 use instant::Instant;
-use libp2p_core::upgrade::{InboundUpgrade, OutboundUpgrade, UpgradeInfo};
-use libp2p_core::Multiaddr;
+use libp2p_core::upgrade::{InboundUpgrade, OutboundUpgrade, Protocol};
+use libp2p_core::{Multiaddr, ToProtocolsIter};
 use libp2p_identity::PeerId;
 use quick_protobuf::{BytesReader, Writer};
-use std::{borrow::Cow, convert::TryFrom, time::Duration};
+use std::{convert::TryFrom, time::Duration};
 use std::{io, iter};
 use unsigned_varint::codec;
 
 /// The protocol name used for negotiating with multistream-select.
-pub const DEFAULT_PROTO_NAME: &[u8] = b"/ipfs/kad/1.0.0";
+pub const DEFAULT_PROTO_NAME: Protocol = Protocol::from_static("/ipfs/kad/1.0.0");
```

Comment:
Note that these are breaking changes, i.e. visible on the outside.

Also, unless we get https://github.com/libp2p/rust-libp2p/pull/3715 merged first, this requires a bump of the `libp2p-core` import in `libp2p-kad` `Cargo.toml`.

Reply:
I am planning to merge #3715 first yes, simply to make all the dependency bumps easier.